app/models/batch_context.rb

```diff
@@ -174,5 +172,9 @@ def verify_staging_location_path
 
     errors.add(:staging_location, 'not a sub directory of allowed parent directories.')
   end
+
+  def verify_file_manifest_exists
+    errors.add(:staging_location, "missing or empty file manifest: #{file_manifest_path}") unless File.exist?(file_manifest_path) && !File.empty?(file_manifest_path)
+  end
```

Comment:
Verify that we at least have a (non-empty) file_manifest before creating the job, because that is fast...actual further validation happens once we instantiate and load the data in the `FileManifest` object (in the job, because that can be slow)